shared/providers.ts

```diff
@@ -286,8 +290,15 @@ export function createReferencesProvider(
             const { repo } = parseGitURI(new URL(textDocument.uri))
 
             let lsifResults: sourcegraph.Location[] = []
-            for await (const lsifResult of lsifProvider(textDocument, position, context)) {
-                if (nonEmpty(lsifResult)) {
+            for await (const rawResult of lsifProvider(textDocument, position, context)) {
+                if (nonEmpty(rawResult)) {
+                    const lsifResult = asArray(rawResult).map(
+                        result =>
+                            (({
+                                ...result,
+                                aggregableTags: [{ text: 'semantic', linkURL: HoverAlerts.linkURL }],
+                            } as any) as sourcegraph.Location)
```

Comment:
are both of these needed? :thinking:

Reply:
Yes right now this is v. v. stupid (there's no type intersection so you have to upcast to top before downcasting to sourcegraph.Location).

Once we publish the type def in the sister PR we won't need any casting.